middle_end/flambda/simplify/basic/simplify_named_result.ml

```diff
@@ -0,0 +1,62 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                   Mark Shinwell, Jane Street Europe                    *)
+(*                                                                        *)
+(*   Copyright 2021 Jane Street Group LLC                                 *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+[@@@ocaml.warning "+a-4-30-40-41-42"]
+
+open! Simplify_import
+
+type descr =
+  | Zero_terms
+  | Single_term of Bindable_let_bound.t * Simplified_named.t
+  | Multiple_bindings_to_symbols of Symbol.t Var_in_binding_pos.Map.t
```

Comment:
Is there a reason to use a map instead of a list here? It seems like we only ever append elements to it and iterate over it. Or is a map needed to avoid duplicates inserted by [simplify_and_lift_set_of_closures]?

Reply:
I just prefer to use maps by default when there is no ordering constraint on an association list.

Reply:
Interesting, I always prefer to use a list if possible. If the order doesn't matter and you aren't going to lookup any entries, then why go to the effort of sorting it.